bin/install

```diff
@@ -110,9 +110,17 @@ get_platform() {
 get_arch() {
   local -r machine="$(uname -m)"
   if [[ $machine == "arm64" ]] || [[ $machine == "aarch64" ]]; then
-    echo "arm64"
+    if [[ "$(get_platform)" != "darwin" ]]; then
+      echo "arm64"
+    else
+      echo "amd64"
+    fi
   elif [[ $machine == *"arm"* ]] || [[ $machine == *"aarch"* ]]; then
-    echo "arm"
+    if [[ "$(get_platform)" != "darwin" ]]; then
+      echo "arm"
+    else
+      echo "amd64"
+    fi
```

Comment:
If this change is what is wanted (see comment first), this should be able to get cleaned up by checking the platform in `if`/`elif` instead of nested `if`s. eg

```bash
  if [[ $machine == "arm64" ]] || [[ $machine == "aarch64" ]] && [[ "$(get_platform)" != "darwin" ]]; then
     echo "arm64"
  elif [[ $machine == *"arm"* ]] || [[ $machine == *"aarch"* ]] && [[ "$(get_platform)" != "darwin" ]]; then
     echo "arm"
```
might also be a tad nicer to pass the platform to the function instead of calling it twice as well.

Reply:
I've incorporated both suggestions. Thanks.